### PR TITLE
NAS-107305 / 12.0 / SNMP iftop

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/freenas/usr/local/bin/snmp-agent.py [hunk 1]
```diff
@@ -527,6 +550,126 @@ def run(self):
...
+                                else:
+                                    data.append(line)
+                    finally:
+                        p.wait()
+                        print(f"iftop for {self.interface} returned {p.returncode}")
+                finally:
+                    with contextlib.suppress(OSError):
+                        os.close(master)
+                    with contextlib.suppress(OSError):
+                        os.close(slave)
```
Comment (william-gr) [inline src/freenas/usr/local/bin/snmp-agent.py:641]: You have already closed slave before fdopen?

File: src/freenas/usr/local/bin/snmp-agent.py [hunk 2]
```diff
@@ -527,6 +550,126 @@ def run(self):
...
+            except Exception as e:
+                print(f"Failed to run iftop for {self.interface}: {e!r}")
+
+            time.sleep(10)
+
+    def _handle_data(self, data):
+        stats = []
+        for line1, line2 in zip(data[::2], data[1::2]):
```
Comment (william-gr) [inline src/freenas/usr/local/bin/snmp-agent.py:649]: Interesting functionality, I wasnt aware of `::` :P

**Review by william-gr — COMMENTED** [review]
Nice job!

File: src/freenas/usr/local/bin/snmp-agent.py (hunk 1 above)
Comment (themylogin) [reply src/freenas/usr/local/bin/snmp-agent.py:641]: That code won't be reached if `subprocess.Popen` fails

File: src/freenas/usr/local/bin/snmp-agent.py (hunk 2 above)
Comment (themylogin) [reply src/freenas/usr/local/bin/snmp-agent.py:649]: I think I use this for the first time in my life.

**Review by william-gr — APPROVED** [review]

**Pull request merged**